tests/test_call_policies.cpp

```diff
@@ -46,6 +46,7 @@ TEST_SUBMODULE(call_policies, m) {
     class Parent {
     public:
         Parent() { py::print("Allocating parent."); }
+        Parent(const Parent& parent) = default;
```

Comment:
It only cared about having these two added, rather than all five. Since it's only in the tests, I wasn't worried about following the rule of five or checking to see which ones were okay to be deleted instead of defaulted.

Reply:
We used to have an explicit default constructor and destructor. The copy constructor (and copy assignment operator) was (were) implicitly defined, which raised a deprecation warning. Conclusion:

- These classes have never had the move constructor or the move assignment operator.
- To restore the old implicit behaviour, we should default the copy assignment operator as well.

Our options, as I see it:

- Match the old implicit behaviour and default the copy assignment operator too.
- Obey the rule of 5 and default the move constructor and move assignment.
- Just ignore this, since these are only tests and not user facing code.

Feel free to take any action you like.

Reply:
I'm deleting the copy assignment, because we don't use it. We need an explicitly defaulted constructor because we have a destructor and we are default constructing these. We aren't using moves, and I think compilers only warn about the old rule of 3 and not the new rule of 5, so for now, I think this is fine (warning driven development is more acceptable for tests than for library code ;) ). I can only detect the cases that fail the warning, so finding all move constructors/assignments would be tricky, anyway.